.github/workflows/rust.yml

```diff
@@ -0,0 +1,29 @@
+name: Test and Build
+
+on: [push, pull_request]
+
+jobs:
+
+  test:
+    runs-on: ubuntu-20.04
+    strategy:
+      fail-fast: false
+      matrix:
+        rust: [stable, nightly, 1.46.0]
+
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions-rs/toolchain@v1
+        with:
+          toolchain: ${{ matrix.rust }}
+          override: true
```

Comment:
Would be better to add `rust-toolchain` file and turn this off but I'm not sure if it plays nicely with matrix.